### ISSUE TITLE
[Bug]: Script fails if no enviroment variable is set

### ISSUE THREAD

**Issue by felipefoz** [CICDToolbox/json-lint#2]
### What happened?

Hi,

Running the script straight forward does not work.

./pipeline.sh: line 118: exclude_list[@]: unbound variable

To make it work I need to set any value in the env. variable.

### Expected behavior

Script should work without setting the Environment Variable.

### How do we reproduct the bug?

Using bash in OS X.

Run the script in a folder with a json file.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by TGWolf** [comment]
I cannot replicate this issue - what version of bash are you running?

It was written and tested on bash 5 (of a Mac) - and the GitHub actions all use the latest ubuntu image.

I vaguely remember issues with the pre-installed version of bash I had on my development Mac and switched to the GNU bash from homebrew.

**Comment by TGWolf** [comment]
Ok, I just located the original version of bash on my Mac (/bin/bash which is version 3.2.57) and now I can replicate the issue.

However, I have also in the same place found my notes on the issue, so this is a bash issue with bash pre 4.2.

Arrays that are defined within a function are not available outside of the function unless you declare with a -g (bash 4.2 ) or by default in bash 5+

I will look into other options, maybe defining the arrays globally at the top and then using them instead of defining them within the function might work.

**Comment by TGWolf** [comment]
Ok, a little more digging and the issue seems to be a little different to what I had in my notes.

bash 3 needs

exclude_array=( )

instead of

declare -a exclude_array=()

I can add in some code which does a work round based on bash version I think, give me 24 hours to to some testing and I will get a new version out.

This issue actually will impact all of the repos in this org as they are based on the same basic pipeline

**Comment by TGWolf** [comment]
Temporary working around:

 EXCLUDE_FILES="nothing" ./pipeline.sh 

This will set the exclude files array and allow the script to run cleanly.

**Comment by TGWolf** [comment]
Possible solution - going to run a battery of tests

    if [[ -n "${EXCLUDE_FILES-}" ]]; then
        IFS=',' read -r -a exclude_list <<< "${EXCLUDE_FILES}"
        echo " Excluded: ${EXCLUDE_FILES}"
        parameters=true
    else
        if ((BASH_VERSINFO[0] < 4)); then
            exclude_list=("")
        else
            # shellcheck disable=SC2034
            declare -a exclude_list=()
        fi
    fi

basically bash 3 does not see an empty array as a bound variable so it is failing, this adds an array with an empty string, it also doesn't like the declare -a hence that being missing to, brief testing seems to suggest this will solve the issue.

**Comment by felipefoz** [comment]
> Ok, I just located the original version of bash on my Mac (/bin/bash which is version 3.2.57) and now I can replicate the issue.
> 
> However, I have also in the same place found my notes on the issue, so this is a bash issue with bash pre 4.2.
> 
> Arrays that are defined within a function are not available outside of the function unless you declare with a -g (bash 4.2 ) or by default in bash 5+
> 
> I will look into other options, maybe defining the arrays globally at the top and then using them instead of defining them within the function might work.

I can confirm, this is the version I have by default (3.2). I also installed bash 5, and in this version it worked flawlessly.

**Comment by TGWolf** [comment]
New version has been pushed to master which should now support base 3.

**Comment by felipefoz** [comment]
Well done. Thanks.
